pkg/services/ngalert/state/cache.go

```diff
@@ -60,7 +60,19 @@ func (c *cache) getOrCreate(ctx context.Context, alertRule *ngModels.AlertRule,
 	}
 
 	if state, ok := c.states[alertRule.OrgID][alertRule.UID][id]; ok {
-		// Annotations can change over time for the same alert.
+		// Annotations can change over time for the same alert, however we don't
+		// want to remove private annotations from previous states as these should
+		// be maintained across evaluations
+		for k, v := range state.Annotations {
```

Comment:
I think we need to be more precise here because nothing stops user from declaring an annotation that starts with `__` and ends with `__`.
I would recommend using this map to check whether an annotation is private
https://github.com/grafana/grafana/blob/952cb4fc0bbbe7159fc4117cca24307acb7d461e/pkg/services/ngalert/models/alert_rule.go#L77-L81